bodhi/server/services/releases.py

```diff
@@ -324,6 +324,17 @@ def save_release(request):
             log.info("Editing release: %s" % edited)
             r = request.db.query(Release).filter(Release.name == edited).one()
             for k, v in data.items():
+                # We have to change updates status to obsolete
+                # if state of release changes to archived
+                if k == "state" and v == ReleaseState.archived and \
+                        r.state != ReleaseState.archived:
+                    updates = request.db.query(Update).filter(Update.release_id == r.id).filter(
+                        Update.status.notin_(
+                            [UpdateStatus.obsolete, UpdateStatus.stable, UpdateStatus.unpushed]
+                        )
+                    ).all()
+                    for u in updates:
+                        u.status = UpdateStatus.obsolete
```

Comment:
The ```Update``` model has a dedicated helper method for setting this state:

```suggestion
                        u.obsolete(request.db)
```

There are two problems with this suggestion, however:

* It causes the request to take a long time. I tried it on the F28 release on my laptop and it took 6 minutes to finish the request. Two solutions I can think of are to add a fedmsg consumer to watch for changes to releases and do this there (in the background), or to add a task worker process to Bodhi that does this with something like Celery. The latter solution is something I've been wanting to do for a while because I don't love the fedmsg solution since it loses messages, but it's also kinda unrelated to this change.
* It untags the builds in Koji. I'm actually not sure if this is a problem or not - I think we should ask infra and releng what they think. @nirik or @puiterwijk or @mohanboddu: is it OK if we untag builds that didn't make it to stable in koji from releases when they become archived at EOL?

Reply:
CC @mohanboddu

Reply:
To clarify a bit, by untag, I mean this function:

https://github.com/fedora-infra/bodhi/blob/3.11.2/bodhi/server/models.py#L3198-L3215

It should only remove tags that Bodhi knows about, which are tags returned from this function:

https://github.com/fedora-infra/bodhi/blob/3.11.2/bodhi/server/models.py#L929-L953

Reply:
The only thing is, if we untag a build, there is a chance that it might get untagged from all tags, if and when it happens the build might get garbage collected. Even then I dont think there would be problem as we garbage collect lot of other builds.

So, I think it should be fine. But, I would like to get other's opinions as well.

Reply:
OK, now that I've had time to think about this some more, I'm convinced that we should not use the obsoletion function and the way you've done it here is good.

One thing I think we should do, however, is add a comment on each update when we change it to obsoleted state. Something like this:

```
u.comment('This update is marked obsolete because the {} release is archived.'.format(u.release.name), agent='bodhi')
```